data-loader/core/src/main/java/com/scalar/db/dataloader/core/util/KeyUtils.java

```diff
@@ -0,0 +1,63 @@
+package com.scalar.db.dataloader.core.util;
+
+import com.scalar.db.api.TableMetadata;
+import com.scalar.db.common.error.CoreError;
+import com.scalar.db.dataloader.core.ColumnKeyValue;
+import com.scalar.db.dataloader.core.exception.Base64Exception;
+import com.scalar.db.dataloader.core.exception.KeyParsingException;
+import com.scalar.db.io.Column;
+import com.scalar.db.io.DataType;
+import com.scalar.db.io.Key;
+import javax.annotation.Nullable;
+
+/** Utility class for creating and dealing with ScalarDB keys. */
+public final class KeyUtils {
+
+  private KeyUtils() {
+    // restrict instantiation
+  }
+
+  /**
+   * Convert a keyValue, in the format of <key>=<value>, to a ScalarDB Key instance.
+   *
+   * @param columnKeyValue A key value in the format of <key>=<value>
+   * @param tableMetadata Metadata for one ScalarDB table
+   * @return A new ScalarDB Key instance formatted by data type
+   * @throws KeyParsingException if there is an error parsing the key value
+   */
+  public static Key parseKeyValue(
+      @Nullable ColumnKeyValue columnKeyValue, TableMetadata tableMetadata)
+      throws KeyParsingException {
+    if (columnKeyValue == null) {
+      return null;
+    }
+    String columnName = columnKeyValue.getColumnName();
+    DataType columnDataType = tableMetadata.getColumnDataType(columnName);
+    if (columnDataType == null) {
+      throw new KeyParsingException(
+          CoreError.DATA_LOADER_INVALID_COLUMN_KEY_PARSING_FAILED.buildMessage(columnName));
```

Comment:
I'm wondering what if there are 3 tables that have `name` column and this exception is thrown for the second table. Users may want to know which table...? What do you think?

Reply:
Updated the method and now passing in the table name as well. I had to add it because the table name is not part of the TableMetadata.

Reply:
@ypeckstadt Thanks for improving the error message!

But, sorry, I should've noticed this when posting the previous comment, other error messages (e.g., DATA_LOADER_INVALID_COLUMN_NON_EXISTENT) have the same issue. Also, the namespace name should be added as well. I think we have 2 options as follows:
1. Add namespace and table names to all the error messages if needed
2. The class is a util class for keys, so it may be okay to include only key name in the error messages from `KeyUtils`. Instead, we'll catch exceptions from `KeyUtils` and wrap it with the namespace and table names like this:
```java
  try {
    :
    // This exception doesn't contain the namespace or table name.
    Key key = KeyUtils.parseKeyValue(columnKeyValue, tableMetadata);
  } catch (KeyParsingException e) {
    // Add the namespace and table names information.
    throw new KeyParsingException(
       CoreError.DATA_LOADER_XXXX_FAILED.buildMessage(namespaceName, tableName), e);
  }
```

I think either is fine. What do you think?

Reply:
Thanks for the improvement!

It looks like there is room to improve the other error messages from the same perspective (i.e., missing information about namespace and table). What do you think?

Reply:
My apologies for the late reply. (had to focus on ScalarFlow more). I have updated the PR and made sure that each exception message includes the namespace and table name.

To avoid having to pass in all 3 with all methods, I have added a wrapper class called `ColumnInfo` that includes all 3 fields.